packages/teuchos/core/src/Teuchos_RCPNode.cpp

```diff
@@ -396,7 +380,16 @@ void RCPNodeTracer::printActiveRCPNodes(std::ostream &out)
       // debug-mode build.
       typedef std::vector<VoidPtrNodeRCPInfoPair_t> rcp_node_vec_t;
       rcp_node_vec_t rcp_node_vec(rcp_node_list()->begin(), rcp_node_list()->end());
-      std::sort(rcp_node_vec.begin(), rcp_node_vec.end(), RCPNodeInfoListPred());
+      std::sort(rcp_node_vec.begin(), rcp_node_vec.end(),
+          [] (const rcp_node_list_t::value_type &v1, const rcp_node_list_t::value_type &v2)
+          {
+#ifdef TEUCHOS_DEBUG
+            return v1.second.nodePtr->insertion_number() < v2.second.nodePtr->insertion_number();
+#else
+            return v1.first < v2.first;
+#endif
+          }
+      );

```

Comment:
So it complains about a function option but not a lambda function?

Reply:
But I like how much more compact the code is.  I am going to love C++11 (now that I am actually allowed to start using it).

Reply:
Yea, I think this is a CUDA quirk: the CUDA compiler compiles the predicate class but not the actual code that uses it to sort later on. Thats my best guess, it definitely looks weird to me. Anyway, I agree, the lambda is a win either way.